lte/gateway/python/integ_tests/s1aptests/test_stateless_multi_ue_mixedstate_mme_restart.py

```diff
@@ -0,0 +1,292 @@
+            print("************************* Resuming Attach procedure "
+                  "UE id ", ue_ids[i+num_attached_ues])
+            for step in range(stateof_ues_in_attachproc_before_restart[i],num_of_steps):
+                if attach_steps[step] == self.exec_sec_mode_complete_step:
+                    tId[ue_ids[i+num_attached_ues]] = attach_steps[step](ue_ids[i+num_attached_ues])
+                elif attach_steps[step] == self.exec_esm_inf_req_step:
+                    attach_steps[step](ue_ids[i+num_attached_ues], tId[ue_ids[i+num_attached_ues]])
+                else:
+                    attach_steps[step](ue_ids[i+num_attached_ues])
+
+        # Verify steady state flows in Table-0
+        # Idle users will have paging rules installed
+        # Active users will have tunnel rules
+        # 1 UL flow is created per active bearer
+        num_ul_flows = num_ues_active + num_ues_attaching
+        # Verify if flow rules are created
+        self._s1ap_wrapper.s1_util.verify_flow_rules(
+            num_ul_flows, self.dl_flow_rules
+        )
+        # Verify paging flow rules for idle sessions
+        self._s1ap_wrapper.s1_util.verify_paging_flow_rules(idle_session_ips)
+
+        # try to bring idle mode users into active state
+        for i in range(num_ues_idle):
+            print(
+                "************************* Sending Service Request ",
+                "for UE id ", ue_ids[i])
+            # Send service request to reconnect UE
+            ser_req = s1ap_types.ueserviceReq_t()
+            ser_req.ue_Id = ue_ids[i]
+            ser_req.ueMtmsi = s1ap_types.ueMtmsi_t()
+            ser_req.ueMtmsi.pres = False
+            ser_req.rrcCause = s1ap_types.Rrc_Cause.TFW_MO_DATA.value
+            self._s1ap_wrapper.s1_util.issue_cmd(
+                s1ap_types.tfwCmd.UE_SERVICE_REQUEST, ser_req
+            )
+            response = self._s1ap_wrapper.s1_util.get_response()
+            self.assertEqual(
+                response.msg_type, s1ap_types.tfwCmd.INT_CTX_SETUP_IND.value
+            )
+            self.dl_flow_rules[idle_session_ips[i]] = []
+
+        # Verify default bearer rules
+        self._s1ap_wrapper.s1_util.verify_flow_rules(
+            tot_num_ues, self.dl_flow_rules
+        )
+
+        # detach everyone
+        print("*** Starting Detach Procedure for all UEs ***")
+        for ue in ue_ids:
+            print("************************* Detaching "
+                  "UE id ", ue)
+            # Now detach the UE
+            detach_req = s1ap_types.uedetachReq_t()
+            detach_req.ue_Id = ue
+            detach_req.ueDetType = (
+                s1ap_types.ueDetachType_t.UE_SWITCHOFF_DETACH.value
+            )
+            self._s1ap_wrapper._s1_util.issue_cmd(
+                s1ap_types.tfwCmd.UE_DETACH_REQUEST, detach_req
+            )
+            # Wait for UE context release command if not in idle mode
+            if ue > num_ues_idle:
```

Comment:
Why are we only checking for UEs not in idle mode? All the UEs are active at this point in the test.

Reply:
Looks like I forgot the update this. In a previous version of the code, I was not moving the idle mode UEs into active state. Thanks for the catch.